frontend/src/components/pages/VerifyEmail.tsx

```diff
@@ -26,18 +83,44 @@ const VerifyEmail = (): React.ReactElement => {
     }
   }, [authenticatedUser, setAuthenticatedUser, refetch]);
 
+  useEffect(() => {
+    const onButtonPress = async () => {
+      if (authenticatedUser && hasBeenPressed) {
+        try {
+          const result = await sendEmailVerificationLink({
+            variables: { email: authenticatedUser.email },
+          });
+          if (result) {
+            const success = result.data?.sendEmailVerificationLink ?? null;
+            if (success) {
+              // TODO: Add rate limiting for button presses
+              setHasBeenPressed(true);
+              return;
+            }
+            setHasError(true);
+          }
```

Comment:
Currently, if there's no `result`, this code won't set an error or success. Is that intended? If not, you can probably combine these checks into one simplified check to reduce nesting.

Reply:
Was a mental carryover from how I did it in AuthAPIClient, will fix.